src/main/java/org/openstreetmap/atlas/utilities/configuration/MergedConfiguration.java

```diff
@@ -81,29 +83,42 @@
 
     private final List<Configuration> configurations;
 
-    public MergedConfiguration(final Resource first, final Iterable<Resource> configurations)
+    public MergedConfiguration(final Configuration... configurations)
     {
-        final LinkedList<Configuration> mergedConfigurations = new LinkedList<>();
-        Iterables.stream(join(first, configurations)).map(StandardConfiguration::new)
-                .forEach(mergedConfigurations::addFirst);
-        this.configurations = Collections.unmodifiableList(mergedConfigurations);
+        this(Arrays.asList(configurations));
     }
 
     public MergedConfiguration(final List<Configuration> configurations)
     {
         this.configurations = Collections.unmodifiableList(configurations);
     }
 
-    public MergedConfiguration(final Configuration... configurations)
+    public MergedConfiguration(final Resource first, final Iterable<Resource> configurations)
     {
-        this(Arrays.asList(configurations));
+        final LinkedList<Configuration> mergedConfigurations = new LinkedList<>();
+        Iterables.stream(join(first, configurations)).map(StandardConfiguration::new)
+                .forEach(mergedConfigurations::addFirst);
+        this.configurations = Collections.unmodifiableList(mergedConfigurations);
     }
 
     public MergedConfiguration(final Resource first, final Resource... configurations)
     {
         this(first, Iterables.iterable(configurations));
     }
 
+    @Override
+    public Set<String> configurationDataKeySet()
+    {
+        // merge the keysets of the underlying StandardConfigurations
+        final Set<String> keySet = new HashSet<>();
+        this.configurations.forEach(configuration ->
+        {
+            keySet.addAll(configuration.configurationDataKeySet());
```

Comment:
Nit: I don't think we want to handle this, but this potentially has unexpected behavior in the case of two underlying configurations having the same key and us merging them into one.

Reply:
If two underlying configurations have the same key, then there is only one value returned for that key, so merging makes sense to me looking at the configuration level.

Reply:
This is a good point. I have not grokked the `MergedConfiguration` class fully enough to understand what happens when you have two underlying configurations with duplicate top level keys. What occurs, in that case, when calling `configuration.get("duplicateTopLevelKey")`? The answer to this question will determine what sort of behavior the `MergedConfiguration#configurationDataKeySet` method should exhibit. Maybe @jklamer or @davis20 can chime in here with some clarification?

To be honest, my initial thought was to leave this method unimplemented for `MergedConfiguration` and just throw a `CoreException("Not implemented.")`. Especially since this method is only being added to `StandardConfiguration` to support my very specific use case. I added this slipshod implementation to `MergedConfiguration` for consistency, but perhaps it might be better to let @jklamer or @davis20 implement this method for `MergedConfiguration` in another PR - since they know/use this class much better than I.

Reply:
The number of keys returned here is the the number of top level endpoints available(from multiple configurations if needed). If two stacked configurations have the same key, then the value on the higher level configuration is returned and the lower configuration value isn't even available through the interface. This implementation makes sense given the design of the interface and what is exposed to the user. 

For example, if two configurations have a "key1", and we returned two copies of that "key1", it would be disingenuous, because calling configuration.get("key1").value() twice would still only return the value of the higher level configuration.

Reply:
I've added a doc comment to the `configurationDataKeySet` method in `MergedConfiguration`, just to make sure the behaviour is absolutely clear to the user.